src/lib.rs

```diff
@@ -162,7 +162,11 @@ where
         while buf_left > 0 {
             let nb_bytes_readable = self.tokens_available().0.unwrap().min(buf_left);
             if nb_bytes_readable < readlimit.min(buf_left) {
-                std::thread::sleep(*window_time);
+                if let Some(lrc) = self.last_read_check {
+                    std::thread::sleep((*window_time - lrc.elapsed()).max(Duration::from_nanos(0)));
```

Comment:
The problem is that last_read_check isn't updated in case we don't have the size available multiple time because there is a `continue` just above and so you will take the time elapsed with the last success read but not the latest try of read.